flang/lib/Lower/Bridge.cpp

```diff
@@ -2265,9 +2277,9 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     // The front-end is currently not adding module variables referenced
```

Comment:
Do we just want to implement what this comment is suggesting at this point? Would it make all this easier if they were just resolved in the front end?

Reply:
We should discuss that with Peter. It may be easier for lowering, though my experience with trying to enforce the equivalent for host associated symbols in internal procedures is that it is hard to enforce this is done properly in name resolution (i.e. they are several places where this creation must happen, and from a front-end perspective, it is not needed so not properly enforced).
The most robust alternative I can think of would be to collect all the symbols referred inside a procedure body when building the PFT instead of using the scopes. But this would add an extra deep pass over the parse-tree.

Reply:
Indeed. Let's perform some due diligence here to cover the bases.